src/uthash.h

```diff
@@ -26,9 +26,11 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 #define UTHASH_VERSION 2.0.1
 
+#ifndef uthash_noheaders
```

Comment:
Macros in general should be ALL_CAPS; uthash_malloc and uthash_free are lowercase because they're supposed to act like (and therefore look like) malloc and free. So uthash_memcmp and uthash_strlen should also look like memcmp and strlen respectively; but UTHASH_FREESTANDING_IMPLEMENTATION should be uppercase by default because there's no reason to make it not-uppercase.

Re the name, I *think* I like `UTHASH_FREESTANDING_IMPLEMENTATION`, but maybe that conflicts with your statement that `<stdint.h>` isn't available on your platform. http://stackoverflow.com/questions/13710101/the-iso-c-standard-defines-two-classes-of-conforming-implementation claims that `<stdint.h>` is available on freestanding implementations as of C99 (but possibly not before).